network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -426,6 +480,86 @@ def get_context(self, request, *args, **kwargs):
         )} - {self.title}'''
         return context
 
+    def serve(self, request, *args, **kwargs):
+        # In Wagtail we use the serve() method to detect POST submissions.
+        # Alternatively, this could be a routable view.
+        if request.body and request.method == "POST":
+            # If the request is POST. Parse the body.
+            data = json.loads(request.body)
+            # If the POST body has a productID and value, it's someone voting on the product
+            if data['productID'] and data["value"]:
+                # Product ID and Value can both be zero. It's impossible to get a Page with ID of zero.
+                product_id = int(data['productID'])  # ie. 68
+                value = int(data["value"])  # ie. 0 to 100
+
+                if value < 0 or value > 100:
+                    # Malicious voting attempt
+                    return HttpResponse('Cannot save vote', status=405, content_type='text/plain')
```

Comment:
There is an dedicated [HttpResponseNotAllowed](https://docs.djangoproject.com/en/2.2/ref/request-response/#django.http.HttpResponseNotAllowed) that we should use instead.